docs/user-guide.rst

```diff
@@ -315,10 +315,34 @@ dictionary in the ``fields`` argument provided to
         "https://httpbin.org/post",
         fields={"field": "value"}
     )
-    
+
     print(resp.json()["form"])
     # {"field": "value"}
 
+If you have a large amount of data, you may prefer to avoid loading all of
+that into memory. To do so, urllib3 provides a ``MultipartEncoder`` that
+handles generating the request data on demand without loading everything in
+memory.
+
+.. code-block:: python
+
+    import urllib3
+    import urllib3.multipart
+
+    encoder = urllib3.multipart.MultipartEncoder({
+      "field": "value",
+      "myfile": ("filename.txt", open("filename.txt", "rb"), "text/plain"),
+    })
+    resp = urllib3.request(
+        "POST",
+        "https://httpbin.org/post",
+        body=encoder,
+        headers=encoder.headers,
+    )
+
+    print(resp.json()["form"])
+    # {"field": "value", "myfile": "..."}
```

Comment:
Another note to investigate: if I run this locally, I just get `{"field": "value"}` as the output; `filename.txt` is uploaded as a file rather than a form-encoded field. 🤔